src/pages/core-candy-machine/guides/create-a-core-candy-machine-ui.md

```diff
@@ -0,0 +1,789 @@
+
+```ts
+import { 
+  findAssetMintCounterPda,
+  fetchAssetMintCounter
+ } from "@metaplex-foundation/mpl-core-candy-machine";
+
+const pda = findAssetMintCounterPda(umi, {
+  id: 1, // The assetMintLimit id you set in your guard config
+  asset: asset.publicKey, // The address of the core nft your user owns
+  candyGuard: candyMachine.mintAuthority,
+  candyMachine: candyMachine.publicKey,
+});
+
+const assetMintCounter = fetchAssetMintCounter(umi, pda);
+```
+
+#### `Allocation` Accounts
+For the `Allocation` guard it can make sense to fetch the `AllocationTracker` to verify that more NFTs can be minted from a given group.
+
+The following code snippet demonstrates how to fetch the `AllocationTracker`. Note that this example assumes you've already obtained the Candy Machine data:
+
+```ts
+import {
+  safeFetchAllocationTrackerFromSeeds,
+} from "@metaplex-foundation/mpl-core-candy-machine";
+
+const allocationTracker = await safeFetchAllocationTrackerFromSeeds(umi, {
+  id: 1, // The allocation id you set in your guard config
+  candyMachine: candyMachine.publicKey,
+  candyGuard: candyMachine.mintAuthority,
+});
+```
+
+#### `Allowlist` Accounts
+When implementing the Allowlist guard, it's crucial to execute a `route` instruction beforehand. This instruction generates a unique account for each wallet and Candy Machine combination, effectively marking the wallet as authorized to mint.
+
+From a UI perspective, it's beneficial to query this account. This allows you to determine whether the `route` instruction needs to be executed or if the user can proceed directly to the mint instruction.
+
+The following code snippet demonstrates how to fetch this account. It assumes that you've already retrieved the Candy Machine data. However, if you prefer, you can hardcode the `candyGuard` and `candyMachine` public keys instead.
+
+```ts
+import {
+  safeFetchAllowListProofFromSeeds,
+  getMerkleRoot,
+} from "@metaplex-foundation/mpl-core-candy-machine";
+
+const allowlist = [
+  "Tes1zkZkXhgTaMFqVgbgvMsVkRJpq4Y6g54SbDBeKVV",
+  "GjwcWFQYzemBtpUoN5fMAP2FZviTtMRWCmrppGuTthJS",
+  "AT8nPwujHAD14cLojTcB1qdBzA1VXnT6LVGuUd6Y73Cy"
+];
+
+const allowListProof = await safeFetchAllowListProofFromSeeds(umi, {
+  candyGuard: candyMachine.mintAuthority,
+  candyMachine: candyMachine.publicKey,
+  merkleRoot: getMerkleRoot(allowlist),
+  user: umi.identity.publicKey,
+});
+```
+
+### Fetch Wallet Data
+To validate the legibility you may also want to fetch information about the connected wallet. Depending on the Guards you are using you may want to know how much SOL is in the wallet and which Tokens and NFTs they own.
+
+To fetch the SOL balance the built in `getAccount` umi function can be used to fetch the wallet account:
+```ts
+const account = await umi.rpc.getAccount(umi.identity.publicKey);
+const solBalance = account.lamports;
```

Comment:
don't know how you want to do this but there's actually a `getBalance` function which might make more sense. Open on this.

```ts
const walletBalance = await umi.rpc.getBalance(umi.identity.publicKey)
const solBalance = walletBalance.basisPoints;
```

Reply:
Mhm, this does not really make a difference, does it? `getBalance` might load less data maybe, but for a normal wallet there should not be much of a difference

Reply:
just terminology. I'm not fussed either way, both works. One function just explains it more explicitly. If it's a deprecated function also no point in changing it.

Reply:
Resolving so that i do not have to retest :)